lib/kubernetes-deploy/runner_task.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+require 'tempfile'
+
+require 'kubernetes-deploy/kubeclient_builder'
+require 'kubernetes-deploy/ui_helpers'
+require 'kubernetes-deploy/kubectl'
+
+module KubernetesDeploy
+  class RunnerTask
+    include KubernetesDeploy::KubeclientBuilder
+    include Kubectl
+
+    class FatalTaskRunError < FatalDeploymentError; end
+
+    def initialize(namespace:, context:, logger: KubernetesDeploy.logger)
+      @logger = logger
+      @namespace = namespace
+      @kubeclient = build_v1_kubeclient(context)
+      @context = context
+    end
+
+    def run(task_template:, command:, args:)
+      validate_configuration(task_template, args)
+
+      raw_template = get_template(task_template, namespace: @namespace)
+      rendered_template = build_pod_template(raw_template, command, args)
+
+      KubernetesDeploy.logger.info("Starting task runner pod #{rendered_template.metadata.name}")
+      @kubeclient.create_pod(rendered_template)
+    end
+
+    private
+
+    def validate_configuration(task_template, args)
+      errors = []
+      if task_template.nil?
+        errors << "Task template name can't be nil"
+      end
+
+      if args.nil?
+        errors << "Args can't be nil"
+      end
+
+      raise FatalTaskRunError, "Configuration invalid: #{errors.join(', ')}" unless errors.empty?
+    end
+
+    def get_template(template_name, namespace:)
+      KubernetesDeploy.logger.info("Fetching task runner pod template: '#{template_name}' in namespace: '#{namespace}'")
+      config_map = @kubeclient.get_config_maps(label_selector: "name=#{template_name}", namespace: namespace).first
+
+      if config_map.nil?
+        raise FatalTaskRunError, "No configmap found matching name=#{template_name} in namespace=#{namespace}"
+      end
+
+      raw = config_map.data.spec
+      template = YAML.load raw
+
+      Kubeclient::Pod.new(template)
+    rescue Psych::SyntaxError => e
+      KubernetesDeploy.logger.error(raw.to_hash)
+      raise FatalTaskRunError, "Task template #{template_name} cannot be parsed: #{e.message}"
+    end
+
+    def build_pod_template(base_template, command, args)
+      KubernetesDeploy.logger.info("Rendering template for task runner pod")
+
+      rendered_template = base_template.dup
+      rendered_template.spec.containers
+        .first.args = args
+
+      container = if rendered_template.spec.containers.length == 1
+        rendered_template.spec.containers.first
+      else
+        rendered_template.spec.containers.find { |cont| cont.name == 'task-runner' }
+      end
+
+      raise FatalTaskRunError, "Pod spec does not contain a templateable container" if container.nil?
+
+      container.command = command
+
+      # TODO: handle environment variables being passed to task
```

Comment:
I've realized since then that they aren't necessary for most of our immediate applications, if an env var is passed to a rake task _after_ the task as an argument rake will take care of the handling automatically.

```
rake task ENV_KEY=VALUE
```
works perfectly.